DoctorWho.web/Controllers/DoctorsController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using AutoMapper;
+using DoctorWho.Db.Repositories;
+using DoctorWho.DTOs.Models;
+using DoctorWho.Web.Validators;
+using FluentValidation.Results;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DoctorWho.web.Controllers
+{
+    [ApiController]
+    [Route("/Doctors")]
+    public class DoctorsController : ControllerBase
+    {
+
+        private readonly DoctorRepository _doctorRepository;
+        private readonly IMapper _mapper;
+        private readonly DoctorDtoValidator _doctorDtoValidtor;
+
+
+        public DoctorsController(DoctorRepository doctorRepository, IMapper mapper, DoctorDtoValidator doctorDtoValidator)
+        {
+            _doctorRepository = doctorRepository;
+            _mapper = mapper;
+            _doctorDtoValidtor = doctorDtoValidator;
+
+
+        }
+
+        [HttpGet]
+        public ActionResult<List<DoctorDto>> GetAllDoctors()
+        {
+            var doctors = _doctorRepository.GetAllDoctors();
+
+            return Ok(doctors.Select(doctor => _mapper.Map<DoctorDto>(doctor)));
+        }
+        [HttpPost]
+        public ActionResult AddDoctor(DoctorDto doctorDto)
+        {
+            ValidationResult validationResult = _doctorDtoValidtor.Validate(doctorDto);
+
+            if (!validationResult.IsValid)
+            {
+                return BadRequest(validationResult.Errors);
+            }
+
+            var doctor = _mapper.Map<Doctor>(doctorDto);
+            _doctorRepository.AddDoctor(doctor);
+            var upsertedDoctorDto = _mapper.Map<DoctorDto>(doctor);
```

Comment:
I don't recommend doing upsert in the POST method
POST is supposed to create resources, if it couldn't (for example due to duplication in the ID) then you should return that as a response error (maybe something like `409 Conflict`)

Reply:
[RFC 7231 - 4.3.3 POST](https://www.rfc-editor.org/rfc/rfc7231#section-4.3.3).

> If one or more resources has been created on the origin server as a
   result of successfully processing a POST request, the origin server
   SHOULD send a 201 (Created) response containing a Location header
   field that provides an identifier for the primary resource created
   ([Section 7.1.2](https://www.rfc-editor.org/rfc/rfc7231#section-7.1.2)) and a representation that describes the status of the
   request while referring to the new resource(s).

The `DoctorDto` doesn't contain the identifier of the created resource (which is the `DoctorId`). The client side application (i.e. front-end) should be able to identify resources so that it can uses endpoints such as `PUT /Doctors/{DoctorId}`